lib/auth/apiserver.go

```diff
@@ -140,6 +140,8 @@ func NewAPIServer(config *APIConfig) http.Handler {
 	srv.DELETE("/:version/tunnelconnections/:cluster/:conn", srv.withAuth(srv.deleteTunnelConnection))
 	srv.DELETE("/:version/tunnelconnections/:cluster", srv.withAuth(srv.deleteTunnelConnections))
 	srv.DELETE("/:version/tunnelconnections", srv.withAuth(srv.deleteAllTunnelConnections))
+	srv.POST("/:version/kube_services", srv.withAuth(srv.upsertKubeService))
```

Comment:
Nit: Not that it matters too much but no other API methods have underscores in them so I'd consider renaming to "kubeservices" for consistency.